improver_tests/blending/weights/test_ChooseWeightsLinear.py

```diff
@@ -420,57 +494,60 @@ def test_above_range_dict(self):
-        )
+@pytest.fixture(
+    params=[set_up_basic_model_config_cube, set_up_basic_model_config_spot_cube]
+)
+def multi_model_inputs(request):
+
+    # create a cube with irrelevant threshold coordinate (dimensions:
+    # model_id: 2; threshold: 2; latitude: 2; longitude: 2)
+    uk_det = request.param(frt=dt(2017, 11, 10), time_points=[dt(2017, 11, 10, 4)])
+    uk_ens = uk_det.copy()
+    uk_ens.coord("model_configuration").points = ["uk_ens"]
+    uk_ens.coord("model_id").points = [2000]
+
+    threshold_cube = iris.cube.CubeList([uk_det, uk_ens]).merge_cube()
+    # create a reference cube as above WITHOUT threshold
+    no_threshold_cube = iris.util.squeeze(threshold_cube[:, 0, :, :].copy())
+    # split into a cubelist by model
+    reference_cubelist = iris.cube.CubeList(
+        [no_threshold_cube[0], no_threshold_cube[1]]
+    )
 
-        # create a reference cube as above WITHOUT threshold
-        self.reference_cube = iris.util.squeeze(self.cube[:, 0, :, :].copy())
+    return threshold_cube, no_threshold_cube, reference_cubelist
 
-        # split into a cubelist by model
-        self.reference_cubelist = iris.cube.CubeList(
-            [self.reference_cube[0], self.reference_cube[1]]
-        )
 
-    def test_slices(self):
-        """Test function slices out extra dimensions"""
-        result = self.plugin._slice_input_cubes(self.cube)
-        self.assertIsInstance(result, iris.cube.CubeList)
-        for cube, refcube in zip(result, self.reference_cubelist):
-            self.assertArrayAlmostEqual(cube.data, refcube.data)
-            self.assertEqual(cube.metadata, refcube.metadata)
-
-    def test_cubelist(self):
-        """Test function creates cubelist with same dimensions where needed"""
-        result = self.plugin._slice_input_cubes(self.reference_cube)
-        self.assertIsInstance(result, iris.cube.CubeList)
-        for cube, refcube in zip(result, self.reference_cubelist):
-            self.assertArrayAlmostEqual(cube.data, refcube.data)
-            self.assertEqual(cube.metadata, refcube.metadata)
-
-    def test_single_cube(self):
-        """Test function populates a cubelist if given a cube with a scalar
-        blending coordinate"""
-        single_cube = self.reference_cube[0]
-        result = self.plugin._slice_input_cubes(single_cube)
-        self.assertIsInstance(result, iris.cube.CubeList)
-        self.assertEqual(len(result), 1)
-        self.assertArrayAlmostEqual(result[0].data, single_cube.data)
-        self.assertEqual(result[0].metadata, single_cube.metadata)
+def test__slice_input_slices(plugin, multi_model_inputs):
+    """Test function slices out extra dimensions to leave only the spatial
+    dimensions. Tested using a cube with and without a threshold coordinate."""
+
+    threshold_cube, no_threshold_cube, reference_cubelist = multi_model_inputs
+
+    for cube in [threshold_cube, no_threshold_cube]:
```

Comment:
This feels like it ought to be parametrized

Reply:
Done.